airbyte/_executors/util.py

```
@@ -148,6 +148,16 @@ def get_connector_executor(  # noqa: PLR0912, PLR0913, PLR0915 # Too many branch
             bool(source_manifest),
         ]
     )
+
+    if version and pip_url:
+        raise exc.PyAirbyteInputError(
+            message="Cannot specify both version and pip_url. Make sure to specify the connector version directly in the pip_url.",
```

Ruff (0.8.2)

Line too long (131 > 100)

(E501)